presto-spi/src/test/java/com/facebook/presto/spi/TestSubfieldTokenizer.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi;
+
+import com.facebook.presto.spi.Subfield.NestedField;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Streams;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestSubfieldTokenizer
+{
+    @Test
+    public void test()
+    {
+        List<Subfield.PathElement> elements = ImmutableList.of(
+                new NestedField("b"),
+                new Subfield.LongSubscript(2),
+                new Subfield.StringSubscript("z"),
+                Subfield.allSubscripts(),
+                new Subfield.StringSubscript("34"),
+                new Subfield.StringSubscript("b \"test\""),
+                new Subfield.StringSubscript("\"abc"),
+                new Subfield.StringSubscript("abc\""),
+                new Subfield.StringSubscript("ab\"cde"));
```

Comment:
I think it worth adding some strange UTF-8 names. Corner cases are emoji, invalid UTF-8, or (valid or invalid) surrogates. For example `"a.b[\"hello\uDBFF\"]"`. There are many examples in `TestStringFunctions`.